resources/assets/js/features/public-profiles/components/Passenger/PassengerProfileContainer.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Preloader from 'app/components/Preloader';
+import PassengerProfile from './PassengerProfile';
+import PassengerAdditionalInfo from './PassengerAdditionalInfo';
+import { getPassengerProfile } from '../../actions';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { getTranslate } from 'react-localize-redux';
+
+import "../../styles/public-profile.scss";
+
+class PassengerProfileContainer extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    componentDidMount() {
+        this.props.getPassengerProfile(this.props.id);
+    }
+
+    render() {
+        const { profile, translate } = this.props;
+
+        if (Object.keys(profile).length === 0) {
```

Comment:
preloader that was before was right solution, but its state should be changed based on request completion or global profile state change